opts/options.go

```diff
@@ -19,13 +19,28 @@ var (
 	DefaultProtocols             = []protocol.ID{ProtocolDHT}
 )
 
+// ModeOpt describes what mode the dht should operate in
+type ModeOpt int
```

Comment:
Created a new mode type that specifies the overall mode the DHT is operating in.